tlvc/fs/block_dev.c

```diff
@@ -222,17 +222,17 @@ static int raw_blk_rw(struct inode *inode, register struct file *filp,
 
 		ebh->b_blocknr = filp->f_pos >> SECT_SIZE_BITS;
 		o_data = bh->b_data;		/* save the 'original' values */
-		o_seg = ebh->b_L2seg;
-		ebh->b_L2seg = current->t_regs.ds;
-		bh->b_data = (unsigned char *)buf;
+		o_seg = (seg_t)ebh->b_L2seg;	/* may be long (xms active) or int */
```

Comment:
If I'm not mistaken, this code stores 32-bits into 16-bits when XMS is active, since oseg is type seg_t (it should be ramdesc_t). In general, it is not recommended to use casts when copying variables unless absolutely necessary as this will hide any compilation warnings that otherwise might result.